doc/release-notes/release-notes-3.1.0-rc1.md

```diff
@@ -0,0 +1,243 @@
+      Fix ordering of transparent outputs such that miner reward is vout[0]
+      Remove assertion that was breaking regtest in the case that blossom activates after the halving.
+      Use for..in rather than an indexed loop.
+      Make evident the relationship between chainparams and key IO.
+      Rename KeyInfo -> KeyConstants and move out of Consensus namespace.
+      Fix typo in constant.
+      Fix assertion check in GetBlockSubsidy
+      Apply suggestions from code review
+      Trivial whitespace fix.
+
+Larry Ruane (7):
+      add python test to reproduce bug 4301
+      flush witness cache correctly
+      review, cleanup: eliminate uninitialized variable
+      self.sync_all(), not time.sleep(4)
+      fix pyflakes CI errors
+      undo flushing witness data on shutdown
+      sync before stopping nodes
+
+Rod Vagg (1):
+      Add joinSplitPubKey and joinSplitSig to RPC
+
+Sean Bowe (23):
+      Add implementations of PRF_expand calls that obtain esk and rcm.
+      Remove bare SaplingNote constructor.
+      Add a getter method to obtain rcm from a Sapling note plaintext.
+      Add support for receiving v2 Sapling note plaintexts.
+      Change transaction builder and miner to use v2 Sapling note plaintexts after Canopy activates.
+      Make ed25519-zebra available via librustzcash.
+      Change to version of ed25519-zebra crate which is compliant with ZIP 215.
+      Enforce ZIP 215 rules upon activation of Canopy.
+      Add test that a weird signature successfully validates.
+      Remove bincode crate.
+      Remove unused curve25519-dalek dev-dependency.
+      Minor adjustments to librustzcash and tests.
+      Redirect git checkouts of ebfull/ed25519-zebra through our vendored sources in offline mode.
+      Require that shielded coinbase output note plaintexts are version 2 if Canopy is active.
+      Make transaction builder take the next block height into account for use of v2 note plaintexts.
+      Turn return values for libsodium-like API into constants for clarity.
+      Add more exhaustive tests for ZIP 215 compatibility.
+      Cargo fmt
+      Remove unused imports from remove_sprout_shielding RPC test.
+      Migrate ZIP 215 test vectors to gtest suite.
+      Change LIBSODIUM_ERROR to -1.
+      Hash "Zcash" to align tests with ZIP 215 test vectors.
+      Remove outdated comment.
+
+Solar Designer (2):
+      Fix typos in ProcessMessage() "headers"
+      During initial blocks download, also report the number of headers
+
+Taylor Hornby (3):
+      Fix undefined behavior in gtest tests
+      Add missing <stdexcept> header for std::invalid_argument
+      Fix bug in CScheduler
+
+Wladimir J. van der Laan (3):
+      Add `-debuglogfile` option
+      test: Add test for `-debuglogfile`
+      doc: Update release notes for `-debuglogfile`
+
+Marshall Gaucher (1):
+      Add helpers for tapping and donating to the testnet faucet
+
+therealyingtong (22):
```

Comment:
Non-blocking (and doesn't need to be done for the RC): add an alias `'therealyingtong': 'Ying Tong Lai'` to `author_aliases` in `zcutil/release-notes.py`.